packages/cli/src/rpc/explain/collect-location-context.ts

```diff
@@ -28,11 +28,15 @@ export default async function collectLocationContext(
 ): Promise<ContextV2.ContextResponse> {
   const result: ContextV2.ContextResponse = [];
 
-  const candidateLocations = new Array<{ location: Location; directory?: string }>();
+  const candidateLocations = new Array<{ location: Location; directory: string }>();
   for (const location of locations) {
     const { path } = location;
     if (isAbsolute(path)) {
       const directory = sourceDirectories.find((dir) => path.startsWith(dir));
+      if (!directory) {
+        warn(`[location-context] Skipping location outside source directories: ${location.path}`);
```

Comment:
Does this mean I can't add external files from me IDE or file system to a Navie chat? 

Since this whole environment is under my control, I am not sure that I would want that.

Reply:
Good point. In general I don't think it's a great idea to allow it to read any file; I definitely don't want the AI to be able to ask for that. If pinned files are handled this way then I'll have to filter any absolute paths in the gatherer, I guess.

Reply:
Fixed in https://github.com/getappmap/appmap-js/pull/2145/commits/5477743e35c40ba7c769214ee45094c28acae24e